sunpy/instr/rhessi.py

```diff
@@ -187,42 +184,43 @@ def get_obssum_filename(time_range):
     Returns
     -------
     out : list
-        Returns the filenames of the observation summary file
+        Returns the urls of the observation summary file
 
     Examples
     --------
     >>> import sunpy.instr.rhessi as rhessi
-    >>> rhessi.get_obssum_filename(('2011/04/04', '2011/04/05'))   # doctest: +SKIP
-    ['http://soleil.i4ds.ch/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits']
-
-    .. note::
-        This API is currently limited to providing data from whole days only.
-
+    >>> rhessi.get_observing_summary_filename(('2011/04/04', '2011/04/05'))   # doctest: +SKIP
+    ['https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110404_042.fits',
+     'https://hesperia.gsfc.nasa.gov/hessidata/metadata/catalog/hsi_obssumm_20110405_031.fits']
     """
     time_range = TimeRange(time_range)
+    # remove time from dates
+    time_range = TimeRange(time_range.start.strftime('%Y/%m/%d'),
+                           time_range.end.strftime('%Y/%m/%d'))
 
-    delta = relativedelta(time_range.end, time_range.start)
-    if delta.years > 0 or delta.months > 0:
-        raise ValueError("Rhessi search results can not be found for a"
-                         " time range crossing multiple months.")
+    filenames = []
 
+    diff_months = (time_range.end.year - time_range.start.year) * 12 +\
+                   time_range.end.month - time_range.start.month
+    first_month = parse_time(time_range.start.strftime('%Y/%m/01'))
+    month_list = list(rrule(MONTHLY, dtstart=first_month, count=diff_months+1))
 
     # need to download and inspect the dbase file to determine the filename
     # for the observing summary data
-
-    dbase_file_name, _ = get_obssumm_dbase_file(time_range)
-    dbase_dat = parse_obssumm_dbase_file(dbase_file_name)
-
-    index_number_start = time_range.start.day - 1
-    # If end is 0 set it to 1 so we always have at least one record.
-    index_number_end = time_range.end.day - 1 or index_number_start + 1
-
-    filenames = dbase_dat.get('filename')[index_number_start:index_number_end]
-    return [posixpath.join(get_base_url(), 'metadata', 'catalog', filename + 's')
-            for filename in filenames]
-
-
-def get_obssumm_file(time_range):
+    # the dbase files are monthly but contain the daily filenames
+    for this_month in month_list:
+        dbase_file_name, hdrs = get_observing_summary_dbase_file(this_month)
+        dbase_dat = parse_observing_summary_dbase_file(dbase_file_name)
+        this_month_obssumm_filenames = dbase_dat.get('filename')
+        daily_filenames_dates = [datetime.strptime(d[0:20], 'hsi_obssumm_%Y%m%d') for d in this_month_obssumm_filenames]
+        for i, this_date in enumerate(daily_filenames_dates):
+            if this_date >= time_range.start and this_date <= time_range.end:
```

Comment:
This is cleaner:
```python
if time_range.start <= this_date <= time_range.end:
```

Reply:
<mind blown>

Reply:
I was totally unaware that Python would resolve such an expression differently from C.